src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.SyntaxReferenceAnalyzerStateData.cs

```diff
@@ -66,6 +73,34 @@ public override void Free()
         }
 
         /// <summary>
+        /// Stores the partial analysis state for operation actions executed on the declaration.
+        /// </summary>
+        internal class OperationAnalyzerStateData : AnalyzerStateData
+        {
+            public HashSet<IOperation> ProcessedOperations { get; set; }
```

Comment:
I copied this from the syntax node logic without thinking deeply. We can guarantee that the set of operations produced for two invocations over the same bound tree will compare equal. I don't know if two invocations over the same symbol will produce the same bound tree, so this needs to be investigated.